src/main/java/com/google/sps/servlets/getBlobKeyServlet.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+ 
+package com.google.sps.servlets;
+ 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.sps.utils.UserUtils; 
+import java.io.*;
+import javax.servlet.*;
+import javax.servlet.http.*;
+ 
+/** Retrieves blobkey from Datastore */
+@WebServlet("/get-blob-key")
+public class getBlobKeyServlet extends HttpServlet {
+  
+  DatastoreService datastore;
+ 
+  public getBlobKeyServlet(){
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+ 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+ 
+    // get user id -- get entity from user id -- new func!!
+    Cookie cookies[] = request.getCookies();
+    Entity userEntity = UserUtils.getUserFromCookie(cookies, datastore);
+ 
+    String blobKey = "noKey";
+    if(userEntity != null){
+      blobKey = (String) userEntity.getProperty("blobKey");
+    }
+    if(blobKey == null){
+        // if the user hasn't uploaded a picture yet, nothing is printed
+        blobKey = "noKey";
+    }
```

Comment:
This is a good opportunity to send an HTTP error response instead of completing the request as normal. Liyah learned how to do this a few weeks ago, so ask her. I forget how.

Pick the most applicable code from this list: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes -- it's a fun read IMO

Reply:
The way I have this error set up is connected to the javascript function where it's called, instead of getting errors when there isn't a picture saved, it just doesn't print a picture. Because this function can easily be called if the user doesn't have a picture, this seemed better; I can make a function to hide the button but that might be a stretch goal

Reply:
A response code is a more robust way to signal the error to the javascript, regardless of how the javascript will handle the error. Please use a response code.